src/Apps/Order/Components/AddressVerificationFlow.tsx

```diff
@@ -152,60 +187,81 @@ const AddressVerificationFlow: React.FC<AddressVerificationFlowProps> = ({
     }
   }, [addressOptions])
 
-  const suggestedAddresses =
-    verifyAddress.suggestedAddresses ||
-    ([] as NonNullable<
-      AddressVerificationFlow_verifyAddress$data
-    >["suggestedAddresses"])
-  const inputAddress = verifyAddress.inputAddress as AddressVerificationFlow_verifyAddress$data["inputAddress"]
-  const verificationStatus = verifyAddress.verificationStatus as AddressVerificationFlow_verifyAddress$data["verificationStatus"]
+  const error = (verifyAddress.verifyAddressOrError as VerifyAddressErrorType)
+    ?.mutationError
+  const suggestedAddresses = (verifyAddress.verifyAddressOrError as VerifyAddressSuccessType)
+    ?.suggestedAddresses
+  const inputAddress = (verifyAddress.verifyAddressOrError as VerifyAddressSuccessType)
+    ?.inputAddress
+  const verificationStatus = (verifyAddress.verifyAddressOrError as VerifyAddressSuccessType)
+    ?.verificationStatus
+
+  const userId = user?.id
+
+  const hasError = Boolean(error)
 
   useEffect(() => {
     const inputOption: AddressOption = {
       ...(inputAddress as any),
-      key: "userInput",
+      key: USER_INPUT,
       recommended: false,
     }
 
-    if (verificationStatus === "VERIFIED_NO_CHANGE") {
-      const verifiedBy = AddressVerifiedBy.ARTSY
-      onChosenAddress(verifiedBy, inputOption.address as AddressValues, true)
+    if (hasError) {
+      const fallbackOption = fallbackFromFormValues(verificationInput)
+      setAddressOptions([fallbackOption])
+      setModalType(ModalType.REVIEW_AND_CONFIRM)
+      trackEvent({
+        action_type: "validationAddressViewed",
+        context_module: ContextModule.ordersShipping,
+        context_page_owner_type: OwnerType.ordersShipping,
+        context_page_owner_id: contextPageOwnerSlug,
+        user_id: userId,
+        flow: "user adding shipping address",
+        subject: "Check your delivery address",
+        option: "review and confirm",
+      })
```

Comment:
I think this is a great way of handling backend errors for this flow.

One suggestion I would make is to differentiate the tracking event from the regular `REVIEW_AND_CONFIRM` instance so that we can tell how many are caused by poor user input v.s. backend errors.

Reply:
Agreed!

Reply:
I take this back. I think that we should act like nothing happened if a backend error is raised.

When this error occurs, we haven't been able to make any determination about the quality of a user's input, so showing the `REVIEW_AND_CONFIRM` dialog is equivalent to us telling a buyer that we don't recognize a legitimate address.

In that scenario I think it's better to abort the address verification flow and allow the user to take their chances with the address they've entered.